tests/test_sim_results/electric_blue/test_volume_timeseries.py

```diff
@@ -0,0 +1,203 @@
+            (DateTime(2020, 2, 29, 0, 20).start_of("year"), DateTime(2020, 2, 29, 0, 20)),
+        ], resolution=AggregationResolution.RES_1_YEAR)
+
+    def test_add_asset_time_series_for_1_year_resolution(self, forward_time_series):
+        volume_time_series = AssetVolumeTimeSeries(
+            "UUID", 5, resolution=AggregationResolution.RES_1_YEAR)
+
+        for market_type, asset_stats in forward_time_series:
+            volume_time_series.add(asset_stats, market_type)
+
+        self.check_time_slots_year(volume_time_series)
+        assert volume_time_series._asset_volume_time_series_buffer == {
+            DateTime(2020, 1, 1, 0, 0, 0): {
+                DateTime(2020, 1, 1, 0, 0, 0): {
+                    "SSP": 36164.701938689024,
+                    "YEAR_FORWARD": {
+                        "bought_kWh": sum_of_energy(
+                            DateTime(2020, 1, 1, 0, 0, 0),
+                            DateTime(2020, 1, 1, 0, 0, 0).end_of("year"),
+                            1
+                        ),
+                        "sold_kWh": sum_of_energy(
+                            DateTime(2020, 1, 1, 0, 0, 0),
+                            DateTime(2020, 1, 1, 0, 0, 0).end_of("year"),
+                            1
+                        )},
+                    "MONTH_FORWARD": {
+                        "bought_kWh": sum_of_energy(
+                            DateTime(2020, 1, 1, 0, 0, 0),
+                            DateTime(2020, 1, 1, 0, 0, 0).end_of("month"),
+                            1
+                        ),
+                        "sold_kWh": sum_of_energy(
+                            DateTime(2020, 1, 1, 0, 0, 0),
+                            DateTime(2020, 1, 1, 0, 0, 0).end_of("month"),
+                            1
+                        )
+                    },
+                    "WEEK_FORWARD": {
+                        "bought_kWh": sum_of_energy(
+                            DateTime(2020, 1, 1, 0, 0, 0),
+                            DateTime(2020, 1, 1, 0, 0, 0).end_of("week"),
+                            1
+                        ),
+                        "sold_kWh": sum_of_energy(
+                            DateTime(2020, 1, 1, 0, 0, 0),
+                            DateTime(2020, 1, 1, 0, 0, 0).end_of("week"),
+                            1
+                        )},
+                    "DAY_FORWARD": {"bought_kWh": sum_of_energy(
+                            DateTime(2020, 1, 1, 0, 0, 0),
+                            DateTime(2020, 1, 1, 0, 0, 0).end_of("day"),
+                            1
+                        ), "sold_kWh": sum_of_energy(
+                            DateTime(2020, 1, 1, 0, 0, 0),
+                            DateTime(2020, 1, 1, 0, 0, 0).end_of("day"),
+                            1
+                        )}}},
+            DateTime(2019, 1, 1, 0, 0, 0): {
+                DateTime(2019, 1, 1, 0, 0, 0): {
```

Comment:
I am really confused now. Why is there an entry for 2019 here and only for the WEEK_FORWARD?

Reply:
I've explained it here: https://github.com/gridsingularity/gsy-framework/pull/393#discussion_r983067160